packages/gatsby/src/redux/actions/add-page-dependency.js

```diff
@@ -36,3 +36,38 @@ exports.createPageDependency = ({ path, nodeId, connection }) => {
   const action = actions.createPageDependency({ path, nodeId, connection })
   store.dispatch(action)
 }
+
+/**
+ * A Graphql resolver middleware that runs `resolver` and creates a
+ * page dependency with the returned node.
+ *
+ * @param resolver A graphql resolver. A function that take arguments
+ * (node, args, context, info) and return a node
+ * @returns A new graphql resolver
+ */
+function pageDependencyResolver(resolver) {
+  return async (node, args, context = {}, info = {}) => {
+    const { path } = context
+    const result = await resolver(node, args, context, info)
+
+    // Call createPageDependency on each result
+    if (path) {
+      const asArray = _.isArray(result) ? result : [result]
+      for (const node of asArray) {
+        // using module.exports here so it can be mocked
+        module.exports.createPageDependency({
```

Comment:
we need to check if node is defined here - `null` can be passed and in that case we definitely don't want to try to create dependency as it will fail on accessing `node.id`